### ISSUE TITLE
What's the status of TeleHash?

### ISSUE THREAD

**Issue by ppetr** [telehash/telehash.org#188]
I apologize that this is probably not the best place to ask questions. But the web page says

> Questions can be filed as issues, asked via twitter, or to Jeremie Miller directly.

and neither channels seem to be appropriate/give any response.

Furthermore the website says

> A version 3 of the protocol is being actively stabilized with multiple interoperable implementations in development. 

but there seems to be no activity on the GitHub page in the past year.

I'd be quite interested in the protocol, as it seems to solve some interesting use-cases for my projects, so it's a pity it seems to be losing interest of its developers.

Another question, are there some examples how to use the library, and/or API documentation? The specification is very thorough, but it's aimed at implementers, not actual users, and it's difficult to figure out how to correctly use the library.

**Comment by quartzjer** [comment]
All of the main activity this past year has been here: https://github.com/telehash/telehash-c/graphs/contributors

As various early deployments of v3 progressed in different use cases, one has become predominately powerful above all others as a core networking platform for IoT with [Filament](http://filament.com). 

The dust hasn't settled there yet either, hence why there isn't much activity on the main telehash.org/etc areas.  There's been a regular stream of new learnings as the IoT use case expands, and I'm expecting a v4 draft with that focus yet this year, as well as the telehash-c codebase being the primary implementation as it's become even easier to bind it into other languages now that it's embedded friendly and standalone.

If your projects are IoT related then you're on the right track and join the [telehash slack team](http://6.telehash.org:3000) as well as take a look at telehash-c and file issues there if it's close to what you need :)

**Comment by ppetr** [comment]
@quartzjer  Thank you for your thorough answer. Indeed, my use-case is IoT related, I want to connect some measurement devices to securely connect and share, regardless of their network location.

Are there any examples available? I imagine something like building a minimalistic client-server or p2p application that sends some sample data etc.

**Comment by NIXOYE** [comment]
@quartzjer any update on this yet? wondering if there is already a way i can start testing telehash

**Comment by quartzjer** [comment]
yes actually, it's my holiday project to get things all reset towards a v4 revamp of the protocol and c code, so stay tuned :)

**Comment by NIXOYE** [comment]
@quartzjer will any gui be available for v4?

**Comment by quartzjer** [comment]
v4 will emerge just as a protocol update and C reference library that is focused on IoT/embedded use cases.

**Comment by NIXOYE** [comment]
Ok thats good too, i want to use telehash on chips and pocketchips for
affordable meshnets

**Comment by isso** [comment]
Hi @quartzjer, there hasn't been any updates for almost a year regarding the v4 spec and the new IoT focused C library. Is there any ETA?

This project is very interesting, and I'm always wondering about its production readiness.

**Comment by quartzjer** [comment]
It's definitely still the plan, unfortunately slow to make it happen but there's many forces wanting this to exist before the end of the year :)

**Comment by isso** [comment]
Great news! is there any draft or something so we can take an earlier look? or even a list of the proposed changes?

**Comment by quartzjer** [comment]
Only informally, but the goal is to move the entire protocol to use the IETF [JOSE standards](https://tools.ietf.org/wg/jose/).  Handshakes will be a JWE wrapped by a JWS, and individual messages will be a JWE with a streaming cipher derived from the handshake.

All of it will support using [JSCN](https://github.com/quartzjer/JSCN) also such that everything remains IoT friendly.

**Comment by RichardWeiYang** [comment]
Is the slack channel still available?

I can't access the auto invitation 

http://6.telehash.org:3000/

**Comment by streaps** [comment]
How do I join the slack channel?

**Comment by Erudition** [comment]
Sadly there's still no activity I guess :(
